File: src/test/java/com/example/pages/AccountCreatedPage.java
package com.example.pages;

import io.qameta.allure.Step;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;

/**
 * Page Object for /account_created – account creation confirmation.
 * Contains: Account Created!, congratulations message and Continue button.
 */
public class AccountCreatedPage extends BasePage {

    private final By accountCreatedHeading =
            By.xpath("//*[contains(.,'Account Created') or contains(.,'ACCOUNT CREATED')]");
    private final By continueButton =
            By.xpath("//*[contains(.,'Account Created') or contains(.,'ACCOUNT CREATED')]//a[contains(.,'Continue') or contains(.,'CONTINUE')]");

    public AccountCreatedPage(WebDriver driver) {
        super(driver);
    }

    public boolean isDisplayed() {
        return isElementPresent(accountCreatedHeading);
    }

    public boolean isAccountCreatedMessageVisible() {
        try {
            wait.until(ExpectedConditions.visibilityOfElementLocated(accountCreatedHeading));
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    @Step("Click Continue after account created")
    public void clickContinue() {
        clickViaJavaScript(continueButton);
    }
}

File: pages/account_created_page.py
"""Account created page – confirmation and Continue button."""
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webdriver import WebDriver
from selenium.webdriver.support import expected_conditions as EC

from pages.base_page import BasePage


class AccountCreatedPage(BasePage):
    account_created_heading = (
        By.XPATH,
        "//*[contains(.,'Account Created') or contains(.,'ACCOUNT CREATED')]"
    )
    continue_button = (
        By.XPATH,
        "//*[contains(.,'Account Created') or contains(.,'ACCOUNT CREATED')]"
        "//a[contains(.,'Continue') or contains(.,'CONTINUE')]"
    )

    def __init__(self, driver: WebDriver):
        super().__init__(driver)

    def is_displayed(self) -> bool:
        return self.is_element_present(self.account_created_heading)

    def is_account_created_message_visible(self) -> bool:
        try:
            self.wait.until(EC.visibility_of_element_located(self.account_created_heading))
            return True
        except Exception:
            return False

    def click_continue(self) -> None:
        self.click_via_javascript(self.continue_button)
